feature-libs/product/multi-dimensional/components/multi-dimensional-container/multi-dimensional-selector/product-multi-dimensional-selector.component.html

```diff
@@ -1,43 +1,72 @@
 <ng-container *ngIf="product">
   <div class="variant-generic-selector">
     <ng-container *ngFor="let category of categories">
-      <label for="{{category.name}}">{{ category.name }}</label>
+      <label>
+        <span>
+          <ng-container
+            *ngIf="
+              getSelectedValue(category.name) as selectedValue;
+              else tmpWithoutLabel
+            "
+          >
+            {{ category.name }}: {{ selectedValue }}
+          </ng-container>
+          <ng-template #tmpWithoutLabel>{{ category.name }}</ng-template>
+        </span>
 
-      <div
-        *ngIf="category.hasImages; else noImages"
-        id="{{category.name}}"
-        class="image-variant-container"
-      >
-        <cx-media
-          *ngFor="let option of category.variantOptions"
-          (keyup.enter)="changeVariant(option.code)"
-          (click)="changeVariant(option.code)"
-          [container]="option.images"
-          [ngClass]="{ active: option?.code === product.code }"
-          [title]="option.value"
-          [attr.aria-label]="option.value"
-          class="image-variant"
-          role="img"
-          tabindex="0"
+        <div
+          *ngIf="category.hasImages; else noImages"
+          role="list"
+          [attr.aria-label]="category.name"
+          class="image-variant-container"
         >
-        </cx-media>
-      </div>
+          <button
+            *ngFor="let option of category.variantOptions"
+            role="listitem"
+            type="button"
+            class="image-variant"
+            [ngClass]="{ active: option?.code === product.code }"
+            [attr.aria-label]="getAriaLabel(option, category.name)"
+            (click)="changeVariant(option.code)"
+            [cxFocus]="{ key: option.code }"
+          >
+            <cx-media
+              [container]="option.images"
+              role="img"
+              [alt]="
```

Comment:
I’m wondering if we should consider using the alt text set in the Backoffice. However, keep in mind that customers might not associate it with this specific multi-dimensional case.

Reply:
Right, I think it should be alt from backoffice.